google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/inspections/AppEngineForbiddenCodeInspection.java

```diff
@@ -66,7 +68,13 @@
     Module module = ModuleUtilCore.findModuleForPsiElement(file);
     final AppEngineStandardFacet appEngineStandardFacet
         = AppEngineStandardFacet.getAppEngineFacetByModule(module);
-    if (appEngineStandardFacet == null || appEngineStandardFacet.isJava8Runtime()) {
+    if (appEngineStandardFacet == null) {
+      return null;
+    }
+
+    Optional<LanguageLevel> runtimeLanguageLevel = appEngineStandardFacet.getRuntimeLanguageLevel();
+    if (!runtimeLanguageLevel.isPresent()
```

Comment:
nit: but I usually like using functional idioms when dealing with Optionals if possible (one of the reasons being to avoid having to do a get())

Reply:
I think this one is unavoidable because we want to short-circuit if the `runtimeLanguageLevel` does not exist or if it's at least 1.8. I'm not sure how to represent that easily via functional methods. (I don't want to wrap the following code in a `.filter(...).ifPresent(() -> { ... })` chain because it would indent a large block of code.

But I used a functional approach in `AppEngineStandardUnsupportedJavaVersionCheck` and `AppEngineJavaeeSupportContributor`, PTAL.